core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html

```diff
@@ -44,14 +80,130 @@
+      <div [hidden]="opportunityDataIsLoading"
+           *ngIf="!translationProgressBar"
+           class="progress">
+        <div class="progress-bar bg-success"
+             role="progressbar"
+             [ngStyle]="progressBarStyle">
+        </div>
+      </div>
+      <div [hidden]="opportunityDataIsLoading"
+           *ngIf="translationProgressBar"
+           class="progress">
+        <div class="progress-bar bg-success"
+             role="progressbar"
+             [ngStyle]="translatedProgressStyle"
+             ngbTooltip="{{opportunity.translationsCount}}/{{opportunity.totalCount}} cards accepted"
+             container="body"
+             placement="bottom">
+        </div>
+        <div class="progress-bar bg-light-green"
+             role="progressbar"
+             [ngStyle]="inReviewProgressStyle"
+             ngbTooltip="{{opportunity.inReviewCount}}/{{opportunity.totalCount}} cards awaiting review"
+             container="body"
+             placement="bottom">
+        </div>
+        <div class="progress-bar bg-light"
+             role="progressbar"
+             [ngStyle]="untranslatedProgressStyle"
+             ngbTooltip="{{cardsAvailable}}/{{opportunity.totalCount}} cards available for translation"
+             container="body"
+             placement="bottom">
+        </div>
+      </div>
+      <lazy-loading [hidden]="!opportunityDataIsLoading"></lazy-loading>
+    </div>
+    <div class="oppia-opportunities-progress-label">
+      <span class="protractor-test-opportunity-list-item-progress-percentage"
+            *ngIf="translationProgressBar"
+            [hidden]="opportunityDataIsLoading"
+            [disableTooltip]="!translationProgressBar"
+            ngbTooltip="Percentage of accepted translated cards"
+            placement="bottom"> ({{ progressPercentage }})
+      </span>
+      <span class="protractor-test-opportunity-list-item-progress-percentage"
+            *ngIf="!translationProgressBar"
+            [hidden]="opportunityDataIsLoading"
+            ngbTooltip="Percentage of accepted questions for skill out of target number of questions per skill ({{targetNumQuestionsPerSkill}})">
+        ({{ progressPercentage }})
+      </span>
+      <lazy-loading [hidden]="!opportunityDataIsLoading"></lazy-loading>
+    </div>
+  </div>
+  <div class="oppia-opportunities-details">
+    <div class="oppia-opportunities-subheading oppia-animate-fadein">
+      <span title="{{ opportunity?.subheading }}"
+            class="protractor-test-opportunity-list-item-subheading oppia-subheading-large"
+            [hidden]="opportunityDataIsLoading">
+        {{ opportunity?.subheading | wrapTextWithEllipsis : 55 }}
+      </span>
+      <span title="{{ opportunity?.subheading }}"
+            class="protractor-test-opportunity-list-item-subheading oppia-subheading-small"
+            [hidden]="opportunityDataIsLoading">
+        {{ opportunity?.subheading | wrapTextWithEllipsis : 20 }}
```

Comment:
The `wrapTextWithEllipsis` might need to be dynamically adjusted based on the breakpoint, since it looks like the text is truncated too early when breakpoint < 500px.

Reply:
As this component is used on multiple pages, setting a value according to a single page might cause issue elsewhere. I have increased the cutoff to 25 (instead of 20) so more of the text is visible without it wrapping over to the next line.